src/main/java/com/google/sps/servlets/MatchingLocationAlgo.java

```diff
@@ -0,0 +1,81 @@
+package com.google.sps.servlets;
+
+
+import com.google.cloud.datastore.*;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.gson.Gson;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.Math.*;
+//import com.google.maps.*;
+//import com.google.android.gms.maps.model;
+//import com.amap.api.maps.model.LatLngBounds;
+//import com.amap.api.maps.model.LatLngBounds;
+//import com.google.code.geocoder-java.*;
+//import org.primefaces.model.map.LatLngBounds;
+
+
+//This could be very useful for turning returned lat/lng into a city name https://developers.google.com/maps/documentation/javascript/examples/geocoding-reverse
+@WebServlet("/MatchingAlgoLocation")
+public class MatchingLocationAlgo extends HttpServlet {
+    //option 2, where we process the center here in the servlet manually using math.
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+        String currEvent = request.getParameter("text-input");
+        Query<Entity> query = Query.newEntityQueryBuilder().setKind("Event").setFilter(PropertyFilter.eq("event-id", currEvent)).build();
```

Comment:
I think it should also be `.setKind("User")`

Reply:
Thank you for reviewing it! I especially appreciate the part about rebasing my branch - that will help a ton!